omod/src/main/webapp/resources/scripts/filters/serverDate.js

```diff
@@ -23,6 +23,27 @@ angular.module('uicommons.filters').
         }
     }])
 
+    /**
+     * This is a function that does the same transformations as `serverDate`, but which renders a date
+     * that is formatted according to the given locale.
+     *
+     * The `options` object is optional. If provided, it should have the format expected by `toLocaleString`; see
+     * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString#using_options.
+     */
+    .filter('serverDateLocalized', ['locale', function(locale) {
+        return function(isoString, options) {
+            console.warn("Use of the `serverDate` and `serverDateLocalized` filters is likely to cause problems. Please consider using the `toLocaleString`, `toLocaleDateString`, or `toLocaleTimeString` filter instead. See Date.prototype.toLocaleString: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString");
+            if (!isoString) {
+                return null;
+            }
+            options = options || { month: 'short' };
+            if (isoString.length > 23) {
+                isoString = isoString.substring(0, 23);
+            }
+            return new Date(isoString).toLocaleString(locale, options);
+        }
+    }])
+
```

Comment:
Not a big deal, and tangential to this PR, but you might want to move, remove/move the methods that do things the "right" way out of this file, since it's called serverDate.js, and make this file about the "bad/dangerous' filters which could eventually be deprecated.

Reply:
I think I did it; let me know if there's anything else I need to do?